common/scripts/poll-npm-package-published.mjs

```diff
@@ -0,0 +1,42 @@
+import { get } from "https";
+import process from "process";
+
+const POLL_INTERVAL_MS = 5 * 1000; // 5 seconds in millis
+const TOTAL_TIME_TO_POLL_MS = 5 * 60 * 1000; // 5 minutes in millis
+const MAX_RETRIES = TOTAL_TIME_TO_POLL_MS / POLL_INTERVAL_MS;
+
+const makeRequest = (url) => {
+  return new Promise((resolve, reject) => {
+    get(url, (response) => {
+      resolve(response);
+    }).end();
+  });
+}
+
+const sleep = (ms) => new Promise( res => setTimeout(res, ms));
+
+const main = async () => {
+  const url = process.argv[2];
+  let retries = 0;
+  while(true) {
```

Comment:
Timeout should be enforced directly instead of by computing number of retries. Something like

```typescript

startTime = new Date();
while (new Date() - startTime < TIMEOUT_MS) {
}
```

Why? Because otherwise you don't account for non-trivial time spent each time in the loop. The `await makeRequest(url)` call below could take 30 seconds. At that point, you're really going to poll for 35 minutes instead of 5 minutes.

Reply:
Great point! Updated to use this!